backend/covexit/api_extensions/views/checkout.py

```diff
@@ -0,0 +1,13 @@
+from oscarapi.views.checkout import OrderDetail as _OrderDetail
+from rest_framework.response import Response
+
+
+class OrderDetail(_OrderDetail):
+    def retrieve(self, request, *args, **kwargs):
+        instance = self.get_object()
+        serializer = self.get_serializer(instance)
+        # once this order has been viewed, mark it "read":
+        if request.user == instance.user:
```

Comment:
We should check if the user is within the user_set of the `Partner` before setting it to read

Reply:
Ah I see where you're coming from. Just to confirm - the idea is that an Order is marked "read" when the company views it, rather than when the user who made the order views it?

Reply:
Exactly

Reply:
After reading through some Oscar source I don't think that will be quite possible - Orders belong to customers but aren't associated with particular Partners. One Order can have many Lines, each one associated with a different Partner. 

We could mark the Order "read" once any/all associated Partners look at it, but I think that's a bit beside the point. Or we could  mark a Line "read' when it's been viewed by its associated Partner.

If you could let me know how the "read" flag was going to be used on the frontend I think that will help determine how to do this. If it's there so that we can send a notification like "Your order has been received", then I think we should add the flag to the Line model.

Reply:
Can't we access the partner via the stock record unit? For now we can assume that one order will always belong to one Partner (and if necessary, enforce it where needed). The UI is this: https://www.figma.com/file/MzDFDSP7oMdUrIZB7fjpVt/Covexit?node-id=197679%3A6940 where as a company you have "new orders" and "order history".
Or under "new orders" we group all the orders that haven't been processed by the company yet. What do you think?

Reply:
@cmazzullo What should we do?

Reply:
Let me get back to you about this.  I think restricting Orders to single Partners and allowing those Partners to directly access the Orders wouldn't be a great idea - Oscar is built around only Users having access to Orders and Partners only having access to Lines. And it would prevent users from being able to buy from multiple sellers with one Basket. 

I think the best solution would be to add a little PartnerOrder model (owned by a partner) to group the Lines for an Order. Don't have time to implement this tonight, will update this branch soon.